### PR TITLE
fix cni configuration creation

### PR DESCRIPTION
**Pull request by tarekhariri93**
I am testing on a Kubernetes cluster deployed with a node which has the control plane, and another node which also has a DPU.

The ovnkube-node on the DPU (in my particular case this is in DPU mode), is not comming up properly, because it is missing its readiness probe: [readiness-probe.go](https://github.com/ovn-org/ovn-kubernetes/blob/12681a716a4c98d8337c7815f15966cc5bba3502/go-controller/cmd/ovn-kube-util/app/readiness-probe.go#L162C1-L171C2),
due to the lack of the CNI configuration file on that particular host.

Looking at the codebase, I saw that the aforementioned CNI configuration is not being written on the DPU because of the following [logic](https://github.com/ovn-org/ovn-kubernetes/blob/12681a716a4c98d8337c7815f15966cc5bba3502/go-controller/pkg/node/default_node_network_controller.go#L1130C2-L1143C4)

### CODE REVIEW

File: go-controller/pkg/node/default_node_network_controller.go [hunk 1]
```diff
@@ -1136,11 +1136,11 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
...
 			return err
 		}
-
-		// Write CNI config file if it doesn't already exist
-		if err := config.WriteCNIConfig(); err != nil {
-			return err
-		}
+	}
+	
+	// Write CNI config file if it doesn't already exist
```
[the remark this answers is not in the document]
Comment (jcaamano) [reply go-controller/pkg/node/default_node_network_controller.go:1141]: cc @wizhaoredhat

**Comment by jcaamano** [comment]
@tarekhariri93 

I think the problem is more on how you are deploying the DPU. The CNI config file should have nothing to do with that readiness probe.

File: go-controller/pkg/node/default_node_network_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tssurya) [reply go-controller/pkg/node/default_node_network_controller.go:1141]: ack that's what I thought considering the conditions seem like we explicitly don't want to write CNI config if its DPU mode.

**Comment by tssurya** [comment]
@tarekhariri93 : Could you close this PR and if you need help figuring out how to deploy the env correctly please provide some details and @wizhaoredhat or @cathy-zhou @girishmg can help you. If you want to talk to us directly to get help, we will be happy to help you - please come to our upstream OVNKube meetings that happen twice a month. Next one is on 5th Feb at 19:00 CET.

**Pull request closed without merging**